### DISCUSSION TITLE
FastApi Users deta support

### DISCUSSION THREAD

**Discussion by Master-Y0da** [fastapi-users/fastapi-users#639] [Ideas]
Since Fastapi adopts [DETA](https://www.deta.sh/) as default and suggested cloud to deploy FastApi apps, would be great if you can add support for this!

**Comment by frankie567** [comment]
I guess you are talking about adding support for "Deta Base" to store users data? If so, I would be open to accept  a PR implementing a database adapter for this database system. It shouldn't be too complex to do.

**Comment by witalijx** [comment]
Are there any updates on this? Also, if you tell me how exactly, I am willing to implement Deta Base Support.

**Reply by frankie567** [reply]
I think nobody tackled the challenge yet 😄

There is a [cookiecutter](https://cookiecutter.readthedocs.io/en/1.7.2/) project which can help you setup a complete project for a database adapter: https://github.com/fastapi-users/fastapi-users-db-cookiecutter

Basically, all you have to do is to run the command:

```sh
cookiecutter https://github.com/fastapi-users/fastapi-users-db-cookiecutter.git
```

and follow the prompts. Then, you'll have a project ready to be completed: you need to implement each methods of the DB adapter class and make the tests working.

Example for MongoDB: https://github.com/fastapi-users/fastapi-users-db-mongodb

**Comment by smorken** [comment]
I started working on this [here](https://github.com/smorken/fastapi-users-db-deta_base).  I just made a new repo since there is no base repo to do a PR into.

It is currently in progress: need to finish a mock Detabase for the tests, and a little polish. I am happy to transfer over the repo or accept any feedback/work on improvements once that's done.

**Comment by smorken** [comment]
Hi there, I think the code [here](https://github.com/smorken/fastapi-users-db-deta_base) is probably ready. The built-in async support for deta Base is forthcoming, so for now it's just implemented with an async [wrapper](https://github.com/smorken/fastapi-users-db-deta_base/blob/master/fastapi_users_db_deta_base/deta_base_async.py)

**Reply by Balogunolalere** [reply]
This is awesome...but It be installed like the rest though...how can I run it

**Reply by frankie567** [reply]
Looks like very nice work @smorken! To be honest, I can't commit to maintain this adapter myself, but if you add some usage documentation and publish it on PyPi, it would be happy to link it in the official documentation :)

**Reply by smorken** [reply]
Thanks! OK, I will eventually add a doc on how to run it, and push to Pypi, but I'll need to work out a few remaining issues as far as integrating with the Deta micro service, so it needs a tiny bit more work. 

I could just keep it on my personal github account, or move it really anywhere else appropriate.  If it stays put, let me know if you'd like to see other changes as far as the links in the README.md file or other files that originated from the [cookiecutter](https://github.com/fastapi-users/fastapi-users-db-cookiecutter)

**Comment by ak4zh** [comment]
I needed deta for one of my project so I did a quick implementation [here](https://github.com/Ak4zh/fastapi_users_db_deta) pushed to pypi.

**Reply by ak4zh** [reply]
Updated to use deta async
